cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -2453,3 +2484,46 @@ def revert_to_snapshot(self, context, volume, snapshot):
             LOG.debug("Backing does not exist for volume.", resource=volume)
         else:
             self.volumeops.revert_to_snapshot(backing, snapshot.name)
+
+    def migrate_volume(self, context, volume, host):
+        """Migrate a volume to the specified host.
+
+        If the backing is not created, the dest host will create it.
+        """
+        backing = self.volumeops.get_backing(volume.name, volume.id)
```

Comment:
Looking at other drivers, they usually check volume status and also if the destination belongs to the same vendor, before trying to migrate stuff.

Reply:
We should also make sure, we have the same capabilities, I guess ... eg. the datastore tag thing we're going to use for slow volumes.

Reply:
we should add location_info to the capabilities reporting in get_volume_stats so we can compare.  We don't want to try and migrate from a non vcenter volume to a vcenter volume. The volume manager can do the generic migration if we simply return False, None.

https://github.com/openstack/cinder/blob/master/cinder/volume/drivers/lvm.py#L759-L769

https://github.com/openstack/cinder/blob/5ee7c1d6cb9775601ec3d3350aff7d1db47612e6/cinder/volume/drivers/lvm.py#L253-L259